core/serialization/deserializer/montage-reviver.js

```diff
@@ -17,6 +17,7 @@ require("../../shim/string");
 
 var PROXY_ELEMENT_MAP = new WeakMap();
 var DATA_ATTRIBUTES_MAP = new Map();
+var MJSON_OBJECTS_MAP = new Map();
```

Comment:
Still don't think this MJSON_OBJECTS_MAP is needed. Since my changes the Deserializer now maintains a cache of its own, and if you asked it to deserialize an object it has already seen it would return the cached version instead of deserializing again. Would that not be good enough for this case?